internal/app/machined/pkg/runtime/platform.go

```diff
@@ -62,3 +64,19 @@ type PlatformNetworkConfig struct {
 
 	Metadata *runtime.PlatformMetadataSpec `yaml:"metadata,omitempty"`
 }
+
+// Equal compares two network configurations.
+func (p *PlatformNetworkConfig) Equal(other *PlatformNetworkConfig) bool {
```

Comment:
probably some tests around this to make sure empty fields are omitted

Reply:
added some unit-tests